Microsoft.Toolkit.Uwp.UI.Animations/ConnectedAnimations/Connected.cs

```diff
@@ -0,0 +1,426 @@
+
+                foreach (var props in _connectedAnimationsProps)
+                {
+                    var connectedAnimation = cas.GetAnimation(props.Key);
+                    var animationHandled = false;
+                    if (connectedAnimation != null)
+                    {
+                        if (props.IsListAnimation && parameter != null && AnimationBase.IsCreatorsUpdateOrAbove)
+                        {
+                            props.ListViewBase.ScrollIntoView(parameter);
+
+                            // give time to the UI thread to scroll the list
+                            var t = props.ListViewBase.Dispatcher.RunAsync(Windows.UI.Core.CoreDispatcherPriority.Normal, async () =>
+                            {
+                                try
+                                {
+                                    var success = await props.ListViewBase.TryStartConnectedAnimationAsync(connectedAnimation, parameter, props.ElementName);
+                                }
+                                catch (Exception)
+                                {
+                                    connectedAnimation.Cancel();
+                                }
+                            });
+
+                            animationHandled = true;
+                        }
+                        else if (!props.IsListAnimation)
+                        {
+                            if (AnimationBase.IsCreatorsUpdateOrAbove && _coordinatedAnimationElements.TryGetValue(props.Element, out var coordinatedElements))
+                            {
+                                connectedAnimation.TryStart(props.Element, coordinatedElements);
+                            }
+                            else
+                            {
+                                connectedAnimation.TryStart(props.Element);
+                            }
+
+                            animationHandled = true;
+                        }
+                    }
+
+                    if (_previousPageConnectedAnimationProps.ContainsKey(props.Key) && animationHandled)
+                    {
+                        _previousPageConnectedAnimationProps.Remove(props.Key);
+                    }
+                }
+
+                // if there are animations that were prepared on previous page but no elements on this page have the same key - cancel
+                foreach (var previousProps in _previousPageConnectedAnimationProps)
+                {
+                    var connectedAnimation = cas.GetAnimation(previousProps.Key);
+                    if (connectedAnimation != null)
+                    {
+                        connectedAnimation.Cancel();
+                    }
+                }
+
+                _previousPageConnectedAnimationProps.Clear();
+            };
+
+            var navigatedPage = NavigationFrame.Content as Page;
+            navigatedPage.Loaded += handler;
```

Comment:
This eventhandler will hold on to an entire page in memory until the event is unhooked. Should be avoided or the very least be weak. If loaded never fires, this stays indefinitely.

Reply:
Same as above, the page can still be cleaned up even if the event is never unregistered because the static class doesn't keep a reference to the page.

Reply:
@skendrot Many reasons. Throws or Collapsed to name a couple. It's not a guarantee